capnp/src/private/arena.rs

```diff
@@ -247,7 +261,12 @@ where
     pub fn into_allocator(self) -> A {
         let mut inner = self.inner.into_inner();
         inner.deallocate_all();
-        inner.allocator.take().unwrap()
+        // See https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/.E2.9C.94.20Extracting.20field.20from.20struct.20that.20implement.20.60Drop.60
+        let (allocator, _segments) = unsafe {
```

Comment:
I'm hesitant to add more `unsafe` like this. I'm tempted to pull in the changes from ##363 *without* the changes from this PR, to see if that still works well. Getting rid of that `RefCell` to me is the big win of all this.

Reply:
As you wish :-) I will likely be testing all those changes in production in the next couple of weeks.

I do think simplifying the codebase, is worth one unsafe block though.

Reply:
https://github.com/capnproto/capnproto-rust/pull/365